modules/mutect2/2.0/mutect2.smk

```diff
@@ -71,6 +77,18 @@ rule _mutect2_get_sm:
         "samtools view -H {input.bam} | grep '^@RG' | "
         r"sed 's/.*SM:\([^\t]*\).*/\1/g'"" | uniq > {output.sm} 2> {log.stderr}"
 
```

Comment:
Could you please add a bit of in-line documentation explaining what this function within a function is doing? It might help for future maintenance.

Reply:
Will do! I straight up stole this from the Manta module, lol.

Reply:
Done!

Reply:
Thanks. Do you know how to get github to allow the changes to show up? Most of the time these resolutions don't appear in the diff.

Reply:
It's probably just my weak Python but what is the second parameter ("input") for and where does it come from?

Reply:
Oh it took me a while to figure this out! `input` is the Snakemake object `input`. So when this function is called by either of the `_mutect2_run` rules, it passes the `wildcards` and `input` objects to this function so that it can parse the `input.candidate_positions` into the command line argument. 

Basically this exists because you can't have a blank input object. So I can't just have `input.candidate_positions` blank in the config and pass it to the shell command, I have to have a test for whether or not the `candidate_positions` file is specified in the config, and then have a dummy VCF in its place in the inputs for the rule, though that dummy VCF doesn't get added to a command-line arg.